2.train-VAE/scripts/6.apply_nmf.py

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Apply NMF to gene dependencies
+# 
+# Non-negative Matrix Factorization (NMF) is a dimensionality reduction technique that factors a non-negative matrix into two non-negative matrices. This is particularly useful when the data is inherently non-negative.
+# 
+# We apply it to GeneEffect scores here and save latent representations (NMF components) for downstream comparative analyses (to compare with BetaVAE).
+
+# In[1]:
+
+
+import sys
+import pathlib
+import pandas as pd
+import plotnine as gg
+from sklearn.decomposition import NMF
+
+sys.path.insert(0, "../utils/")
+from data_loader import load_model_data
+
+
+# In[2]:
+
+
+data_directory = pathlib.Path("../0.data-download/data")
+dependency_file = pathlib.Path(f"{data_directory}/CRISPRGeneEffect.parquet")
+gene_dict_file = pathlib.Path(f"{data_directory}/CRISPR_gene_dictionary.parquet")
+
+
+output_dir = pathlib.Path("results")
+nmf_output_file = pathlib.Path(f"{output_dir}/nmf_latent.parquet.gz")
+output_nmf_weights_file = pathlib.Path(f"{output_dir}/NMF_weight_matrix_gsea.parquet")
+
+
+# In[3]:
+
+
+nmf_components = 50
```

Comment:
Is this dependent on the ICA components? If so consider writing these parameters to a `.json` or `.yaml` for storage and import for consistent parameters across files.